cranelift/codegen/src/isa/aarch64/inst/emit.rs

```diff
@@ -1275,20 +1303,18 @@ impl MachInstEmit for Inst {
             }
             &Inst::AtomicRMW { ty, op } => {
                 /* Emit this:
-                      dmb         ish
                      again:
-                      ldxr{,b,h}  x/w27, [x25]
+                      ldaxr{,b,h}  x/w27, [x25]
                       op          x28, x27, x26 // op is add,sub,and,orr,eor
-                      stxr{,b,h}  w24, x/w28, [x25]
+                      stlxr{,b,h}  w24, x/w28, [x25]
                       cbnz        x24, again
-                      dmb         ish
```

Comment:
Would it make sense to keep ARMv7 support and only emit the new sequence for ARMv8?

Reply:
We certainly will need a similar sequence when we complete the ARM32 (ARMv7) backend, but that doesn't share code with the aarch64 (aka ARMv8) one we're in here, so I think it's fine/expected that we avoid ARMv7-isms here.